### PR TITLE
website: Update input documentation & demos

### PR DESCRIPTION
**Pull request by FlyersPh9**
- Add input demos for the different options.
- Update documentation, include adding hyperlinks to other mentioned components.

### CODE REVIEW

File: apps/website/src/examples/Input.button.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+export default () => {
+  return (
+    <>
+      <LabeledInput
+        label='Address'
+        placeholder='1600 Pennsylvania Avenue, Washington, DC 20500'
+        defaultValue='685 Stockton Dr, Exton, PA 19341'
+        svgIcon={
+          <IconButton styleType='borderless'>
```
Comment (mayank99) [inline apps/website/src/examples/Input.button.tsx:17]: icon buttons should have an [accessible name](https://www.tpgi.com/what-is-an-accessible-name/), which you can specify using `aria-label`

File: apps/website/src/examples/Input.button.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import { LabeledInput, IconButton } from '@itwin/itwinui-react';
+import { SvgCloseSmall } from '@itwin/itwinui-icons-react';
+
+export default () => {
+  return (
+    <>
+      <LabeledInput
+        label='Address'
+        placeholder='1600 Pennsylvania Avenue, Washington, DC 20500'
+        defaultValue='685 Stockton Dr, Exton, PA 19341'
```
[the remark this answers is not in the document]
Comment (FlyersPh9) [reply apps/website/src/examples/Input.button.tsx:15]: I've replaced `Address` with `Street` and used https://www.fakexy.com/ to create a fake value.

File: apps/website/src/examples/Input.button.tsx (hunk 1 above)
Comment (FlyersPh9) [reply apps/website/src/examples/Input.button.tsx]: Added an `aria-label` to the button example.

File: apps/website/src/pages/docs/input.mdx [hunk 3]
```diff
@@ -9,17 +9,80 @@ group: inputs
...
+  <AllExamples.InputButtonExample client:load />
+</LiveExample>
+
+You can pair an input and a button together.
+
+<LiveExample src='ButtonGroup.input.tsx'>
+  <AllExamples.ButtonGroupInputExample client:load />
+</LiveExample>
+
+There are more complex options of pairing inputs with other various form components when using the [button group](buttongroup).
```
[the remark this answers is not in the document]
Comment (elephantcatdog) [reply apps/website/src/pages/docs/input.mdx:84]: I agree. I think it's easier to see the text and example/image as 'together' when the text is above.

**Comment by mayank99** [comment]
There are some merge conflicts.

File: apps/website/src/pages/docs/input.mdx [hunk 4]
```diff
@@ -9,17 +9,80 @@ group: inputs
...
+
+It’s possible to use inputs inline, in forms for example. The status displays a bit differently. For these textareas the status border remains the same, but the icon displays horizontally centered inside the area, at the far right.
+
+### Status
+
+<LiveExample src='Input.status.tsx'>
+  <AllExamples.InputStatusExample client:load />
+</LiveExample>
+
+Status messages are sometimes required for bringing warnings or errors to the user’s attention. For example, required fields that are left empty, spelling mistakes, or to notify a user the username they have entered is available. To achieve this, the bottom border gains the color of the alert level, and the corresponding icon and message displays right below the field, to ensure confusion between errors and text fields is unlikely. Note that the status is conveyed differently when [inline](#inline) styling is applied.
```
[the remark this answers is not in the document]
Comment (FlyersPh9) [reply apps/website/src/pages/docs/input.mdx]: The visual description stuff has been removed.

File: apps/website/src/pages/docs/input.mdx (hunk 3 above)
[the remark this answers is not in the document]
Comment (FlyersPh9) [reply apps/website/src/pages/docs/input.mdx:61]: Can we leave the order as is & then address it all at once when we finish the components?  

Previous PRs I merged were going:
1. Title
2. Paragraph
3. Demo
4. Paragraph
5. Demo
6. etc

But I found myself confusing myself a couple times in scenarios like this screenshot.  That demo is for the previous section, not for Standalone.

Screenshot 2023-04-06 at 4 01 17 PM

So I changed it to:
1. Title
2. Demo
3. Paragraph
4. Demo
5. Paragraph
6. etc.

Maybe the order isn't as much the issue here, maybe the vertical spacing between sections can be larger to make sections more distinct?

File: apps/website/src/pages/docs/input.mdx [hunk 5]
```diff
@@ -9,17 +9,80 @@ group: inputs
...
+
+## Variants
+
+### Input
+
+<LiveExample src='Input.input.tsx'>
+  <AllExamples.InputInputExample client:load />
+</LiveExample>
+
+Use `<Input>` when all that is needed is a text field with no label, hint text, or status. In most scenarios we strongly recommend an input being paired with a label.
```
[the remark this answers is not in the document]
Comment (FlyersPh9) [reply apps/website/src/pages/docs/input.mdx]: I've removed the "In most scenarios" and explicitly stated that a label is an accessibility requirement.  I've also removed variants and added a standalone section.

[the remark this answers is not in the document]
Comment (FlyersPh9) [reply apps/website/src/pages/docs/input.mdx]: Most search inputs don't have an input, does search somehow get an exception?

[the remark this answers is not in the document]
Comment (mayank99) [reply apps/website/src/pages/docs/input.mdx]: If the input has other contextual hints to help the user enter the right text (e.g. a search icon in the case of a search input, then a _visible_ label can be omitted, but there still needs to be a hidden label. This hidden label could be associated using `aria-labelledlby` or `<label>` + `VisuallyHidden` for example.

File: apps/website/src/pages/docs/input.mdx (hunk 3 above)
[the remark this answers is not in the document]
Comment (mayank99) [reply apps/website/src/pages/docs/input.mdx:61]: Can we change both the order and the vertical spacing then?
Doesn't have to be in this PR, but if you're writing new content, why not use the correct order right from the start?

File: apps/website/src/pages/docs/input.mdx (hunk 5 above)
[the remark this answers is not in the document]
Comment (FlyersPh9) [reply apps/website/src/pages/docs/input.mdx]: Good to know, I'll try using `<label>` + `VisuallyHidden` on the search boxes in the demo pages of #1073.

**Review by mayank99 — APPROVED** [review]

File: apps/website/src/pages/docs/input.mdx [hunk 6]
```diff
@@ -9,17 +9,80 @@ group: inputs
...
+
+There are three different sizes, which can be specified using the `size` prop.
+
+### Inline
+
+<LiveExample src='Input.inline.tsx'>
+  <AllExamples.InputInlineExample client:load />
+</LiveExample>
+
+It’s possible to use inputs inline, in forms for example. The status displays a bit differently. For these textareas the status border remains the same, but the icon displays horizontally centered inside the area, at the far right.
```
[the remark this answers is not in the document]
Comment (elephantcatdog) [reply apps/website/src/pages/docs/input.mdx]: I still don't think the visual description is necessary here.

[the remark this answers is not in the document]
Comment (FlyersPh9) [reply apps/website/src/pages/docs/input.mdx]: I thought I did this, my mistake.  Change is made.

**Review by elephantcatdog — APPROVED** [review]

**Pull request merged**